File: app/src/main/java/org/forgerock/auth/ForceAuthRequestInterceptor.java
/*
 * Copyright (c) 2020 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.auth;

import androidx.annotation.NonNull;

import org.forgerock.android.auth.Action;
import org.forgerock.android.auth.Request;
import org.forgerock.android.auth.RequestInterceptor;

import static org.forgerock.android.auth.Action.START_AUTHENTICATE;

/**
 * Sample {@link RequestInterceptor} to add ForceAuth
 */
public class ForceAuthRequestInterceptor implements RequestInterceptor {

    @NonNull
    @Override
    public Request intercept(@NonNull Request request) {
        if (((Action)request.tag()).getType().equals(START_AUTHENTICATE)) {
            return request.newBuilder()
                    .url(request.url().toString() + "&ForceAuth=true")
                    .build();
        }
        return request;
    }
}

File: samples/app/src/main/java/com/example/app/setting/ForceAuthRequestInterceptor.kt
/*
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package com.example.app.setting

import android.net.Uri
import org.forgerock.android.auth.Action
import org.forgerock.android.auth.FRRequestInterceptor
import org.forgerock.android.auth.Request

/**
 * Sample [RequestInterceptor] to add ForceAuth
 */
class ForceAuthRequestInterceptor : FRRequestInterceptor<Action> {
    override fun intercept(request: Request, tag: Action): Request {
        return if (tag.type == Action.START_AUTHENTICATE) {
            request.newBuilder()
                .url(Uri.parse(request.url().toString())
                    .buildUpon()
                    .appendQueryParameter("ForceAuth", "true").toString())
                .build()
        } else request
    }
}
